superset-frontend/src/SqlLab/components/RunQueryActionButton.tsx

```diff
@@ -91,25 +91,32 @@ const RunQueryActionButton = ({
     ? (DropdownButton as React.FC)
     : Button;
 
+  const isDisabled = !sql.trim();
```

Comment:
Nice, this is way more readable now, and the intent is clear!

Reply:
Thank you! Though this was actually Elizabeth's suggestion. So can't take all the credit.